iothub/device/src/RecommendedAction.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Recommended actions for users to take upon different ConnectionStatus and ConnectionStatusChangeReason
+    /// </summary>
+    public enum RecommendedAction
+    {
+        /// <summary>
+        /// It's recommended to take no more actions as all operations on the device or module client will be carried out as normal.
+        /// This is expected when <see cref="ConnectionStatus.Connected"/> and <see cref="ConnectionStatusChangeReason.ConnectionOk"/>.
+        /// </summary>
+        NoActionWhenNormal,
```

Comment:
NoAction seems wrong given the action they should be taking is resuming normal device operations, like sending telemetry.

It is only NoAction with respect to addressing the connection.

Reply:
NoRecommendation?

Reply:
Changed this to `PerformNormally` as per our discussion